src/Text/Pandoc/Class/PandocMonad.hs

```diff
@@ -668,6 +669,29 @@ withPaths (p:ps) action fp =
   catchError ((p </> fp,) <$> action (p </> fp))
              (\_ -> withPaths ps action fp)
 
+-- | If @fp@ doesn't exist, but the data file relative to @/subdir/@
+-- does, then return that; otherwise return @fp@.
+findFile :: PandocMonad m
```

Comment:
Sorry to revise my original suggestion, but what about `findFileWithDataFallback`?
`findFile` should be reserved for something less special-purpose.

Reply:
Also, could we use this function to simplify `readMetadataFile`?

Reply:
Both done